plugins/bufferlines/bufferline.nix

```diff
@@ -125,196 +215,395 @@ in
+      max_prefix_length = defaultNullOpts.mkInt 15 ''
+        Maximum prefix length used when a buffer is de-duplicated.
+      '';
 
-      diagnosticsIndicator = helpers.defaultNullOpts.mkLuaFn null "Either `null` or a function that returns the diagnostics indicator.";
+      sort_by = defaultNullOpts.mkStr "id" ''
+        How to sort the buffers.
 
-      diagnosticsUpdateInInsert = helpers.defaultNullOpts.mkBool true "Whether diagnostics should update in insert mode";
+        Possible values: `insert_after_current`, `insert_at_end`, `id`, `extension`, `relative_directory`, `directory`, `tabs`.
+        Also accepts a function with a signature `function(buffer_a, buffer_b)` allowing you to compare with custom logic.
+      '';
 
-      offsets = helpers.defaultNullOpts.mkNullable (types.listOf types.attrs) null "offsets";
+      diagnostics =
+        defaultNullOpts.mkEnumFirstDefault
+          [
+            false
+            "nvim_lsp"
+            "coc"
+          ]
+          ''
+            Diagnostics provider.
+
+            Set to `false` to disable.
+          '';
+
+      diagnostics_indicator = defaultNullOpts.mkLuaFn null ''
+        Either `null` or a function that returns the diagnostics indicator.
+      '';
+
+      diagnostics_update_on_event = defaultNullOpts.mkBool true ''
+        Use nvim's diagnostic handler.
+      '';
+
+      offsets = defaultNullOpts.mkNullable (types.listOf types.attrs) null "offsets";
 
       groups = {
-        items = helpers.defaultNullOpts.mkListOf types.attrs [ ] "List of groups.";
+        items = defaultNullOpts.mkListOf types.attrs [ ] "List of groups.";
 
         options = {
-          toggleHiddenOnEnter = helpers.defaultNullOpts.mkBool true "Re-open hidden groups on bufenter.";
+          toggle_hidden_on_enter = defaultNullOpts.mkBool true ''
+            Re-open hidden groups on `BufEnter`.
+          '';
         };
       };
 
       hover = {
-        enabled = mkEnableOption "hover";
+        enabled = lib.mkEnableOption "hover";
 
-        reveal = helpers.defaultNullOpts.mkListOf types.str [ ] "reveal";
+        reveal = defaultNullOpts.mkListOf types.str [ ] "reveal";
 
-        delay = helpers.defaultNullOpts.mkInt 200 "delay";
+        delay = defaultNullOpts.mkInt 200 "delay";
       };
 
       debug = {
-        logging = helpers.defaultNullOpts.mkBool false "Whether to enable logging";
+        logging = defaultNullOpts.mkBool false "Whether to enable logging.";
       };
+    };
 
-      customFilter = helpers.defaultNullOpts.mkLuaFn null ''
-        ```
-        fun(buf: number, bufnums: number[]): boolean
-        ```
-      '';
-
-      highlights = genAttrs (attrValues highlightOptions) (
+    highlights =
+      let
+        newHighlightsOptions = map (
+          option:
+          let
+            optionSnakeCase = lib.nixvim.toSnakeCase option;
+          in
+          optionSnakeCase
+        ) (oldHighlightOptions ++ [ "truncMarker" ]);
```

Comment:
I don't think we should be using the _old_ option names here? Firstly it seems wrong that we have to convert them to snake_case, additionally we're delaying the inevitable needing to add a new option without affecting the old option deprecations.

We're already _kinda_ running into this issue with `trunkMarker` vs `trunc_marker`.

Reply:
Yeah, fair enough. My intent was to reduce size of the module with more verbose option declarations. Since its in the `settingsOptions` it allows a user to specify anything not called outright so I wasn't too concerned about adding more options, yet.

But it still does feel like a kicking the can down the road solution...